sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -150,15 +150,15 @@ class EquivalentExpressions(
   //   1. CodegenFallback: it's children will not be used to generate code (call eval() instead)
```

Comment:
I think this is still true. Why do we change the code here?

Reply:
No, after this change, if all the children expressions support code-gen,  we will call code-gen all its children, and then call `nullSafeEval()` to evaluate the current expression logic. Main change code `$objectTerm = (($clazz) references[$idx]).nullSafeEval(${childrenParameter});`
https://github.com/apache/spark/pull/42450/files#diff-855822d5be236405f279207562850e426d17eb0dc3d5c83bd8db72385af602c2R49-R78